Development/nmos/client_utils.cpp

```diff
@@ -55,39 +60,165 @@ namespace nmos
                 }
             };
         }
+#endif
+        // get the associated network interface name from the IP address
+        inline utility::string_t get_interface_name(const utility::string_t& address)
+        {
+            utility::string_t interface_name;
+            if (!address.empty())
+            {
+                const auto interfaces = web::hosts::experimental::host_interfaces();
+
+                auto find_interface = [&]()
+                {
+                    return boost::range::find_if(interfaces, [&](const web::hosts::experimental::host_interface& interface)
+                    {
+                        return interface.addresses.end() != boost::range::find(interface.addresses, address);
+                    });
+                };
+                const auto interface = find_interface();
+                if (interfaces.end() != interface) { interface_name = interface->name; }
+            }
+            return interface_name;
+        }
```

Comment:
Since this function is basically copied from the nmos-cpp-node example code, probably we should make it fit for both uses and put it in _cpprest/host_utils.(h,cpp)_?

This might work?

```suggestion
        // get the associated network interface name from an IP address
        inline utility::string_t get_interface_name(const utility::string_t& address, const std::vector<web::hosts::experimental::host_interface>& host_interfaces = web::hosts::experimental::host_interfaces())
        {
            const auto interface = boost::range::find_if(interfaces, [&](const web::hosts::experimental::host_interface& interface)
            {
                return interface.addresses.end() != boost::range::find(interface.addresses, address);
            });
            return interfaces.end() != interface ? interface->name : utility::string_t{};
        }
```